bin/version-compare.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+version_compare () {
+    if [[ $1 == 'latest' ]] || [[ $1 == 'dev' ]] || [[ $1 == 'nightly' ]]
+    then
+		# Set first arg to a huge number considering latest, dev or nightly to be the highest version.
+        set -- 1000000 "${@:2:3}"
+    fi
+    if [[ $2 == 'latest' ]] || [[ $2 == 'dev' ]] || [[ $2 == 'nightly' ]]
+    then
+		# Set second arg to a huge number considering latest, dev or nightly to be the highest version.
+        set -- "${@:1}" 1000000 "${@:3}"
+    fi
```

Comment:
After debugging it seems to be a bit buggy.

- first you don't pass the 3 arguments to this function but only the 2 first ones passed to the script. So `$@` references the 2 parameters passed to this function.
- in `${@:2:3}` the second value corresponds on the number of values you want to extract from the `$@` arguments list. Otherwise all the arguments will be extracted until the end of the list.

So I suggest something like this
```suggestion
    if [[ $1 == 'latest' ]] || [[ $1 == 'dev' ]] || [[ $1 == 'nightly' ]]
    then
		# Set first arg to a huge number considering latest, dev or nightly to be the highest version.
        set -- 1000000 "${@:2}"
    fi
    if [[ $2 == 'latest' ]] || [[ $2 == 'dev' ]] || [[ $2 == 'nightly' ]]
    then
		# Set second arg to a huge number considering latest, dev or nightly to be the highest version.
        set -- "${@:1:1}" 1000000
    fi
```
In the first condition, as `$2` is the last argument, we don't need the number of arguments to extract.
In the second condition, we only want to extract the first `$1` argument. Otherwise with `${@:1}` all the arguments are extracted.

Reply:
The 3 arguments *are* required for the bash script. The action always pass this arguments, if not then the script isn't used correctly.
The default value of the operator is handle in the action directly, not the script.

Reply:
Yes! I perfectly understand how it works in the action and I tested/debugged the script. I found something went wrong when we pass 'latest' espacially as the second script argument.
Of course we can tallked about it IRL to better understand what I found and I suggested 😉